aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/ApiMocking.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.appsync;
```

Comment:
`AppSyncMocking` lives in `com.amplifyframework.datastore.appsync` since the `AppSync` interface does. Your new `ApiMocking` might be a `testutil`ity, since it mocks a `core` spec (the `ApiCategoryBehavior`.) In this view, it could live somewhere under [`testutils/src/main/java/com/amplifyframework/testutils`](https://github.com/aws-amplify/amplify-android/tree/main/testutils/src/main/java/com/amplifyframework/testutils). I don't feel 100% resolved in this, though. What do you think?